mbed_greentea/mbed_greentea_cli.py

```diff
@@ -371,14 +442,26 @@ def main_cli(opts, args, gt_instance_uuid=None):
         for tid in accepted_target_ids:
             gt_log_tab("accepting target id '%s'"% gt_bright(tid))
 
-    test_exec_retcode = 0
-
     test_exec_retcode = 0       # Decrement this value each time test case result is not 'OK'
     test_platforms_match = 0    # Count how many tests were actually ran with current settings
     target_platforms_match = 0  # Count how many platforms were actually tested with current settings
 
-    test_report = {}    # Test report used to export to Junit, HTML etc...
-
+    test_report = {}            # Test report used to export to Junit, HTML etc...
+    muts_to_test = []           # MUTs to actually be tested
+    test_queue = Queue()        # contains information about test_bin and image_path for each test case
+    test_result_queue = Queue() # used to store results of each thread
+    execute_threads = []        # list of threads to run test cases 
```

Comment:
there is available threadpool, which could be better choose than array of threads